broker/src/test/java/io/camunda/zeebe/broker/system/partitions/AsyncSnapshottingTest.java

```diff
@@ -114,37 +117,38 @@ private void createAsyncSnapshotDirectorOfReplayMode() {
   public void shouldValidSnapshotWhenCommitPositionGreaterEquals() {
     // given
     createAsyncSnapshotDirectorOfProcessingMode();
-    asyncSnapshotDirector.forceSnapshot();
+    final var snapshot = asyncSnapshotDirector.forceSnapshot();
 
     // when
     setCommitPosition(100L);
 
     // then
-    Awaitility.await()
-        .untilAsserted(() -> assertThat(persistedSnapshotStore.getLatestSnapshot()).isPresent());
+    assertThat(snapshot.join()).isNotNull();
+    assertThat(persistedSnapshotStore.getLatestSnapshot()).isPresent();
   }
 
   @Test
   public void shouldTakeSnapshotsOneByOne() {
     // given
     createAsyncSnapshotDirectorOfProcessingMode();
-    asyncSnapshotDirector.forceSnapshot();
+    final var firstSnapshot = asyncSnapshotDirector.forceSnapshot();
     setCommitPosition(99L);
-    Awaitility.await()
-        .untilAsserted(() -> assertThat(persistedSnapshotStore.getLatestSnapshot()).isPresent());
-    final PersistedSnapshot oldSnapshot = persistedSnapshotStore.getLatestSnapshot().orElseThrow();
+    assertThat(firstSnapshot.join()).isPresent();
+    final var firstSnapshotIndex = firstSnapshot.join().orElseThrow().getIndex();
 
     // when
-    asyncSnapshotDirector.forceSnapshot();
+    final var secondSnapshot = asyncSnapshotDirector.forceSnapshot();
     setCommitPosition(100L);
 
     // then
-    Awaitility.await()
-        .untilAsserted(
-            () ->
-                assertThat(persistedSnapshotStore.getCurrentSnapshotIndex())
-                    .describedAs("New snapshot is taken")
-                    .isGreaterThan(oldSnapshot.getIndex()));
+    assertThat(secondSnapshot.join())
+        .describedAs("Second snapshot is taken")
+        .isPresent()
+        .get()
+        .extracting(PersistedSnapshot::getIndex, as(InstanceOfAssertFactories.LONG))
+        .describedAs("Second snapshot has a higher index")
+        .isGreaterThan(firstSnapshotIndex);
```

Comment:
:wrench: Would be also nice to compare the snapshot with the snapshot for the snapshot store. Wdyt?